contracts/BC_fusion/StakeHub.sol

```diff
@@ -0,0 +1,866 @@
+                "the doubleSignSlashAmount is out of range"
+            );
+            doubleSignSlashAmount = newDoubleSignSlashAmount;
+        } else if (_compareStrings(key, "downtimeJailTime")) {
+            require(value.length == 32, "length of downtimeJailTime mismatch");
+            uint256 newDowntimeJailTime = _bytesToUint256(32, value);
+            require(
+                newDowntimeJailTime >= 2 days && newDowntimeJailTime < doubleSignJailTime,
+                "the downtimeJailTime is out of range"
+            );
+            downtimeJailTime = newDowntimeJailTime;
+        } else if (_compareStrings(key, "doubleSignJailTime")) {
+            require(value.length == 32, "length of doubleSignJailTime mismatch");
+            uint256 newDoubleSignJailTime = _bytesToUint256(32, value);
+            require(
+                newDoubleSignJailTime >= 100 days && newDoubleSignJailTime > downtimeJailTime,
+                "the doubleSignJailTime is out of range"
+            );
+            doubleSignJailTime = newDoubleSignJailTime;
+        } else if (_compareStrings(key, "maxEvidenceAge")) {
+            require(value.length == 32, "length of maxEvidenceAge mismatch");
+            uint256 newMaxEvidenceAge = _bytesToUint256(32, value);
+            require(newMaxEvidenceAge >= 7 days, "the maxEvidenceAge is out of range");
+            maxEvidenceAge = newMaxEvidenceAge;
+        } else {
+            require(false, "unknown param");
+        }
+        emit paramChange(key, value);
+    }
+
+    /*----------------- view functions -----------------*/
+    function isPaused() external view returns (bool) {
+        return _stakingPaused;
+    }
+
+    function getEligibleValidators() external view returns (IBSCValidatorSet.Validator[] memory, bytes[] memory) {
+        return (_eligibleValidators, _eligibleValidatorVoteAddrs);
+    }
+
+    function getValidatorBasicInfo(address operatorAddress)
+        external
+        view
+        returns (address consensusAddress, address poolModule, bytes memory voteAddress, bool jailed, uint256 jailUntil)
+    {
+        Validator memory valInfo = _validators[operatorAddress];
+        consensusAddress = valInfo.consensusAddress;
+        poolModule = valInfo.poolModule;
+        voteAddress = valInfo.voteAddress;
+        jailed = valInfo.jailed;
+        jailUntil = valInfo.jailUntil;
+    }
+
+    function getValidatorDescription(address operatorAddress) external view returns (Description memory) {
+        return _validators[operatorAddress].description;
+    }
+
+    function getValidatorCommission(address operatorAddress) external view returns (Commission memory) {
+        return _validators[operatorAddress].commission;
+    }
+
+    function getValidatorWithVotingPower(
+        uint256 offset,
+        uint256 limit
+    ) external view returns (address[] memory consensusAddrs, uint256[] memory votingPowers, uint256 totalLength) {
```

Comment:
I think this is called by consensus engine, you should return the vote address as well, as the validatorset contract will need it.